src/Admin/Product/Attributes/AttributesBlock.php

```diff
@@ -0,0 +1,200 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Admin\Product\Attributes;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\AdminConditional;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Conditional;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Registerable;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\MerchantCenterService;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\AttributeManager;
+use Automattic\WooCommerce\Admin\BlockTemplates\BlockInterface;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class AttributesBlock
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Admin\Product\Attributes
+ */
+class AttributesBlock implements Service, Registerable, Conditional {
+
+	use AdminConditional;
+	use AttributesTrait;
+
+	/**
+	 * @var AttributeManager
+	 */
+	protected $attribute_manager;
+
+	/**
+	 * @var MerchantCenterService
+	 */
+	protected $merchant_center;
+
+	/**
+	 * AttributesBlock constructor.
+	 *
+	 * @param AttributeManager      $attribute_manager
+	 * @param MerchantCenterService $merchant_center
+	 */
+	public function __construct( AttributeManager $attribute_manager, MerchantCenterService $merchant_center ) {
+		$this->attribute_manager = $attribute_manager;
+		$this->merchant_center   = $merchant_center;
+	}
+
+	/**
+	 * Register a service.
+	 */
+	public function register(): void {
+		// compatibility-code "WC >= 8.3" -- The Block Template API used requires at least WooCommerce 8.3
+		if ( ! version_compare( WC_VERSION, '8.3', '>=' ) ) {
+			return;
+		}
+
+		// Register the hooks only if Merchant Center is set up.
+		if ( ! $this->merchant_center->is_setup_complete() ) {
+			return;
+		}
+
+		// https://github.com/woocommerce/woocommerce/blob/8.3.0/plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/AbstractProductFormTemplate.php#L16
+		$template_area = 'product-form';
+
+		// https://github.com/woocommerce/woocommerce/blob/8.3.0/plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php#L361
+		$block_id = 'product-attributes-section';
+
+		add_action(
+			"woocommerce_block_template_area_{$template_area}_after_add_block_{$block_id}",
+			function ( BlockInterface $attributes_section ) {
+				// Please note that the simple and variable product types use the same product block template 'simple-product'.
+				if ( 'simple-product' !== $attributes_section->get_root_template()->get_id() ) {
+					return;
+				}
+
+				$section = $this->add_section( $attributes_section );
```

Comment:
Thanks for the suggestion.

This method call will be inlined by [this commit in PR #2179](https://github.com/woocommerce/google-listings-and-ads/pull/2179/commits/5cf49b154ac52a6adb4f0d8a842ad9adb5117d24#diff-2b96a65cd604d35139576ee14972bfb745506fe4a781d276e9c721d20a18fa59R110-R128). Instead of using `$this` to indirectly add group and section to template, the changed code fragment now calls to the literal callee, which I believe should not cause similar confusion as here.

Reply:
Thanks, yes that will remove the ambiguity!